internal/git/config.go

```diff
@@ -9,16 +9,33 @@ import (
 	"github.com/pkg/errors"
 )
 
+//go:generate moq -out config_mock.go . GitConfigAPI
+
+// GitConfigAPI is an interface to interact with a git config system
+// this makes it possible to extract information from the repository
+// or the local user
+type GitConfigAPI interface {
+	CommitterDetails() (name string, email string, err error)
```

Comment:
It is somewhat unusualy that the provider of functionality also defines its interface? This would normally go into the call site to make sure it only contains what is relevant to the caller instead of all the things a provider provides.

WDYT?

Reply:
I agree. It is usually a good idea to use structs as normal and just wrap it in an interface when needed. I will see what I can do to improve this.

Reply:
Should be moved now, I can see the benefit of this, but maybe it shouldn't be as close to the caller as it is now?